apis/kubedb/v1alpha2/druid_helpers.go

```diff
@@ -511,18 +511,25 @@ func (d *Druid) SetDefaults() {
 		}
 	}
 
-	if d.Spec.MetadataStorage == nil {
-		d.Spec.MetadataStorage = &MetadataStorage{}
-	}
-	if !d.Spec.MetadataStorage.ExternallyManaged {
-		if d.Spec.MetadataStorage.ObjectReference == nil {
-			d.Spec.MetadataStorage.ObjectReference = &kmapi.ObjectReference{}
+	d.SetDefaultsToMetadataStorage()
+	d.SetDefaultsToZooKeeperRef()
+
+	if d.Spec.Monitor != nil {
+		if d.Spec.Monitor.Prometheus == nil {
+			d.Spec.Monitor.Prometheus = &mona.PrometheusSpec{}
 		}
-		d.Spec.MetadataStorage.Name = d.GetMetadataStorageName()
+		if d.Spec.Monitor.Prometheus != nil && d.Spec.Monitor.Prometheus.Exporter.Port == 0 {
+			d.Spec.Monitor.Prometheus.Exporter.Port = kubedb.DruidExporterPort
+		}
+		d.Spec.Monitor.SetDefaults()
 	}
-	if d.Spec.MetadataStorage.Namespace == "" {
-		d.Spec.MetadataStorage.Namespace = d.Namespace
+}
+
+func (d *Druid) SetDefaultsToMetadataStorage() {
+	if d.Spec.MetadataStorage == nil {
+		d.Spec.MetadataStorage = &MetadataStorage{}
 	}
+	d.Spec.MetadataStorage.ObjectReference = d.GetMetadataStorageObjectRef()
 	if d.Spec.MetadataStorage.LinkedDB == "" {
 		d.Spec.MetadataStorage.LinkedDB = "druid"
 	}
```

Comment:
If externallyManaged is true, but metadataStorage.name is set by defaulter. Then no `appBinding` exists.